### ISSUE TITLE
Merge Tabs Components (formerly "Segmented control lacks suitable colour contrast")

### ISSUE THREAD

**Issue by pollecuttn** [wellcomecollection/wellcomecollection.org#10239]
## NEW TICKET

See below for original ticket. This is now a ticket that works on merging SegmentedControl and TabNav into one and aligning their behaviour as far as one of them still behaves as tabs and the other still behaves as anchors. @davidpmccormick started the work in #10424 and left a todo list which I'll share below.

- [x] ~how to handle border beneath~ discussed with @j-jaworski: optionally hide bottom border when e.g. component is flush against a block with background colour
- [ ] get real with colour, space etc. values from [figma file](https://www.figma.com/file/juZQR2wi2BsYG8TDH6R4uk/Wellcome-Collection-%E2%80%93-Styles-and-Component-Library-2023?type=design&mode=design) – largely just eye-balling so far
- [ ] Add Inter Medium and Semi Bold to match the styles
- [ ] fix keyboard focus styles
- [ ] check if we want to keep any hover transition behaviour (currently width animates left-to-right) (@j-jaworski to look)
- [ ] improve Storybook stories
- [x] this isn't specific to the `Tabs` component as the issue already exists on the `whats-on` page (see screenshot below), but do we want to either include a 'no events' message beneath a month tab that doesn't have any events (yet), or maybe don't display the tab whatsoever? ([separate ticket was created](https://github.com/wellcomecollection/wellcomecollection.org/issues/10434))

Raph is continuing the work in #10432 

## ORIGINAL TICKET
**What**
Segmented control lacks suitable colour contrast

**Where**
https://wellcomecollection.org/whats-on

**TODO**

- [x] Timebox investigating swapping out the segmented control for the tabbed nav

**Background**

From the Hassell Inclusion accessibility audit August 2023.

Dovetail: https://proficient-wallaby-dpwd.dovetailapp.com/insights/Segmented-control-keyboard-focus-should-provide-a-continuous-experience-5Yl7Q7lxvWzA3WxB9t4PdM

https://proficient-wallaby-dpwd.dovetailapp.com/insights/Segmented-control-lacks-suitable-colour-contrast-3vMi4tFpmu1gtmb7jLdtDi

Miro board: https://miro.com/app/board/uXjVMjsL5YE=/

**Comment by davidpmccormick** [comment]
We could use the `TabNav` and `SubNavigation` components to accomplish the same behaviour with both links and tabs, and these have possibly clearer accessibility hover/focus states (obviously it would also increase consistency of UI across the site). I think this would need a bit of design direction in terms of spacing etc. if we want to make this change though cc @j-jaworski 

__Links__
<img width="1173" alt="image" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/1394592/3cf5b260-f0db-4233-abf8-5bf95fc2f6c1">

__Tabs__
<img width="362" alt="image" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/1394592/d11b8722-19e5-40f7-b6c5-ca5e9aa09a2f">

**Comment by davidpmccormick** [comment]
Work so far in [`tab-nav-segmented-control` branch](https://github.com/wellcomecollection/wellcomecollection.org/tree/tab-nav-segmented-control)

**Comment by j-jaworski** [comment]
I have reviewed the UI for `Segementedcontrol` and `TabNav` in-line with the buttons audit that had previously been completed
In this work:
- Definition of a tab
- Audit of tab-like components across cardigan and wc.org
- Minor adjustments to the UI to align with tabs component on other products
- Focus state design specified to align with buttons and links work previously completed
- Margin and padding spec
- Text styles and underline spec
- Mock-ups to show how implementing this design would impact key pages


**Designs in Figma**
https://www.figma.com/file/juZQR2wi2BsYG8TDH6R4uk/Wellcome-Collection-%E2%80%93-Styles-and-Component-Library-2023?type=design&node-id=1610%3A11863&mode=dev&t=Ije4xzqFm9BqyAog-1
<img width="849" alt="Screenshot 2023-11-03 at 09 51 03" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/44364115/3e998771-409c-49fa-a85a-a57f0db006f8">


<img width="1446" alt="Screenshot 2023-11-03 at 09 50 52" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/44364115/aa2457c2-227d-42fd-962c-1d0f71e08401">

**Comment by rcantin-w** [comment]
Part of it could be merged so ready for review but will be pulled back once its through as we still need to discuss hover behaviour ( @j-jaworski is having a look)

**Comment by j-jaworski** [comment]
After much messing about, we should keep the hover behaviour as it is but change the colour from pale yellow to Neutral 300.

`background-color: rgb(255, 235, 173);`

Potentially once I can see this in production we might want to change it so that the height, rather than the width is the property that we are animating on hover, but for now lets just keep it as it is.

**Comment by j-jaworski** [comment]
Nice! I think the hover behaviour looks good, couple of Qs
1. Is it possible to update the font weights to align with the designs?
2. At the moment, the focus state is showing when I click on a tab, should it only show if the user is using the focus state to navigate?

**Comment by rcantin-w** [comment]
Oops, I was fixing point 2 as I shared the link, was too keen hehe. Here goes the newest version: https://62f13cdbd0ff140768a8d87b-uaamwhhaen.chromatic.com/?path=/story/components-tabs--basic

For the font weight it's what we discussed yesterday about adding a weight to the whole? Do we have the file on hand, I can find it on Google fonts but we currently seem to use hosted files. 
 
Currently we have two only for Inter:
Regular: 
font-weight: 500; (used to be 400 but we changed it together yesterday. Reckon it needs to go back to 400 [as regular is 400 and medium is 500](https://fonts.google.com/specimen/Inter))
https://i.wellcomecollection.org/assets/fonts/Inter-Regular.woff2

and Bold:
font-weight: 700;
https://i.wellcomecollection.org/assets/fonts/Inter-Bold.woff2

And we're talking about adding Medium? 
(@gestchild fyi)

**Comment by j-jaworski** [comment]
ahh okay, I think we need:
Regular 400
Medium 500 
Semi-bold 600
With the current `Bold 700` changing to `semi-bold 600`. 
We can either do this globally (although I haven't assessed the impact) or host all four font weights and update styles at a granular level.
